### PR TITLE
Fix deploying non-stable channels and explicit revs

### LINKED ISSUE #143: Handling of channels and explicit revs is broken
If you call `model.deploy()` with an explicit, non-stable channel, the stable channel still gets deployed.  If you try to use an explicit rev without specifying the channel, it fails with a "charm not found" error.

### PR DESCRIPTION
**Pull request by johnsca**
Fixes #143

### CODE REVIEW

File: juju/model.py [hunk 1]
```diff
@@ -1072,8 +1072,6 @@ def _get_series(self, entity_url, entity):
                     application_name = entity['Meta']['charm-metadata']['Name']
                 if not series:
                     series = self._get_series(entity_url, entity)
-                if not channel:
-                    channel = 'stable'
```
Comment (tvansteenburgh) [inline juju/model.py]: Why was this removed?

Comment (johnsca) [inline juju/model.py]: Because it's wrong.  ;)  If an explicit rev is given, it may not be the stable channel.  There's also no actual need to provide a value for the channel, as we already resolve the charm to an explicit rev by querying the charm store.

File: tests/integration/test_model.py [hunk 2]
```diff
@@ -36,6 +36,22 @@
 
 @base.bootstrapped
 @pytest.mark.asyncio
+async def test_deploy_channels_revs(event_loop):
```
Comment (wwwtyro) [inline tests/integration/test_model.py]: :heart:

File: juju/model.py (hunk 1 above)
Comment (tvansteenburgh) [inline juju/model.py]: Okay, makes sense, I just wanted to make sure the subsequent call to AddCharm wouldn't fail when passed a None channel.

Comment (johnsca) [inline juju/model.py]: Sure.  The test does cover that case.

Comment (kwmonroe) [inline juju/model.py]: Sorry to hijack this @johnsca, but while you're in here, would you please fix the docstring on line 1005 s/development/edge?

Comment (johnsca) [inline juju/model.py]: Sure.  Done.
